src/Castle.Core/DynamicProxy/ProxyUtil.cs

```diff
@@ -15,15 +15,56 @@
 namespace Castle.DynamicProxy
 {
 	using System;
+	using System.Collections.Generic;
+	using System.Linq;
 	using System.Reflection;
+	using System.Runtime.CompilerServices;
 
-	using Castle.DynamicProxy.Internal;
 #if FEATURE_REMOTING
 	using System.Runtime.Remoting;
 #endif
 
+	using Castle.Core.Internal;
+	using Castle.DynamicProxy.Generators.Emitters;
+
 	public class ProxyUtil
```

Comment:
Let's go ahead and make this static like it should have been, however don't change the functions back to extension methods.